File: src/javascript/ValidAnagram.js
function isAnagram(s, t) {
  if (s.length !== t.length) {
    return false;
  }
  s = s.split("").sort()
  t = t.split("").sort()

  for (let i = 0; i < s.length; i++) {
    if (s[i] !== t[i]) {
      return false;
    }
  }
  return true;
}

let s = "anagram";
let t = "nagaram";
let anagram = isAnagram(s, t);
console.log(anagram);

File: src/ValidAnagram.java
public class ValidAnagram {

  public static void main(String[] args) {
    var s = "anagram";
    var t = "nagaram";
    boolean anagram = isAnagram(s, t);
    System.out.println(anagram);
  }

  public static boolean isAnagram(String s, String t) {
    int a = s.length();
    int b = t.length();
    if (a != b) {
      return false;
    }
    int[] memory = new int[26];
    for (int i = 0; i < a; i++) {
      memory[s.charAt(i) - 'a']++;
      memory[t.charAt(i) - 'a']--;
    }
    for (int i : memory) {
      if (i != 0) {
        return false;
      }
    }
    return true;
  }
}
